docs: add default params to the spec (#2416)
celestiaorg/celestia-app: specs/src/specs/params.md

| auth.TxSizeCostPerByte | 10 | Gas used per transaction byte. | True |
| auth.SigVerifyCostED25519 | 590 | Gas used to verify ED25519 signature. | True |
| auth.SigVerifyCostSecp256k1 | 1000 | Gas used to verify secp256k1 signature. | True |
| bank.SendEnabled | true | Allow transfers. | False |
| blob.GasPerBlobByte | 8 | Gas used per blob byte. | True |
| MaxBlockBytes | 100MB | Hardcoded value in tendermint for the protobuf encoded block. | False |
| MaxSquareSize | 128 | Hardcoded value in the applications that requires a hardfork to change. | False |
| blob.GovMaxSquareSize | 64 | Governance parameter for the  square size. If larger than MaxSquareSize, MaxSquareSize is used. | True |
| consensus.block.MaxBytes | 1.8MB | Governance parameter for the maximum size of the block. | True |
| consensus.block.MaxGas | -1 / ∞ | Maximum gas allowed per block (-1 is infinite). | True |
| consensus.block.TimeIotaMs | 1000 | Minimum time added to the time in the header each block. | False |
| consensus.evidence.MaxAgeNumBlocks | 100000 | The maximum number of blocks before evidence is considered invalid. This value will stop CometBFT from pruning block data. | True |

> | consensus.evidence.MaxAgeDuration | 172800000000000 | The maximum age of evidence before it is considered invalid. This value should be identical to the unbonding period. | True |

Reviewer: Can we move this into a separate issue to not block this PR?

# Celestia Governance Params

These are the parameters for mainnet. Note that not all of these parameters are
changable via governance. This list also includes parameter that require a
hardfork to change due to being manually hardcoded in the application or they
are blocked by the `x/paramfilter` module.

## Parameters

> | Module.Parameter | Default | Summary | Changeable via Governance |

Reviewer: What about https://github.com/celestiaorg/celestia-app/tree/main/x/qgb#params btw?

Author: added here https://github.com/celestiaorg/celestia-app/pull/2416/commits/e8930a7cdb093225c1d4c478f757731ab14f3693

> | Module.Parameter | Default | Summary | Changeable via Governance |

Reviewer: General remark: The `Default` columns should be completely consistent! Currently it is sometimes `0.05 (5%)` (like the Minimum commission rate) but sometimes only `0.02` (like the community pool). 
I personally think the default value should first be in the column in the unit the param is passed and then explained in a more human friendly way. The comment should explicitly mention the unit this is passed into the param. See my comment for MaxAgeDuration for a concrete example: https://github.com/celestiaorg/celestia-app/pull/2416/files#r1315177546

Author: updated mainly here https://github.com/celestiaorg/celestia-app/pull/2416/commits/debdc9b69e250190f3846260b32a91c037916484

| --- | --- | --- | --- |

> | auth.MaxMemoCharacters | 256 | Largest allowed size for a memo. | True |

Reviewer 2: Clarify character encoding.

| auth.TxSigLimit | 7 | Max number of signatures allowed in a multisig transaction. | True |
| auth.TxSizeCostPerByte | 10 | Gas used per transaction byte. | True |

> | auth.SigVerifyCostED25519 | 590 | Gas used to verify ED25519 signature. | True |

Reviewer 2: https://en.wikipedia.org/wiki/EdDSA#Ed25519

```suggestion
| auth.SigVerifyCostED25519 | 590 | Gas used to verify Ed25519 signature. | True |
```

| auth.SigVerifyCostSecp256k1 | 1000 | Gas used to verify secp256k1 signature. | True |
| bank.SendEnabled | true | Allow transfers. | False |
| blob.GasPerBlobByte | 8 | Gas used per blob byte. | True |

> | MaxBlockBytes | 100MB | Hardcoded value in tendermint for the protobuf encoded block. | False |

Reviewer 2: Are all of these `MB` or `MiB`?

> | MaxBlockBytes | 100MB | Hardcoded value in tendermint for the protobuf encoded block. | False |

Reviewer 2: Should this say CometBFT?

```suggestion
| MaxBlockBytes | 100MB | Hardcoded value in CometBFT for the protobuf encoded block. | False |
```

> | MaxSquareSize | 128 | Hardcoded value in the applications that requires a hardfork to change. | False |

Reviewer: Can you add something like (128*128*512bytes=8MiB) or would this be too verbose?

Reviewer 2: IMO not needed since `MiB` is unambiguous.

Author: for square size? atm I just updated to be rows and cols of shares, which I think is more descriptive. We have other descriptions of shares in the `shares.md` and how we encode them which are required to come up with an accurate value

| blob.GovMaxSquareSize | 64 | Governance parameter for the  square size. If larger than MaxSquareSize, MaxSquareSize is used. | True |

> | consensus.block.MaxBytes | 1.88MiB | Governance parameter for the maximum size of the block. | True |

Reviewer: Can you add a detail of what "size of the block" refers to here? could mean EDS/ODS/raw block data in bytes...

Author: :+1: 992fb4e130692c908625940642aba7d63f95d2f4

> | consensus.block.MaxGas | -1 / ∞ | Maximum gas allowed per block (-1 is infinite). | True |

Reviewer 2: This looks like a division.

```suggestion
| consensus.block.MaxGas | -1 | Maximum gas allowed per block (-1 is infinite). | True |
```

| consensus.block.TimeIotaMs | 1000 | Minimum time added to the time in the header each block. | False |
| consensus.evidence.MaxAgeNumBlocks | 100000 | The maximum number of blocks before evidence is considered invalid. This value will stop CometBFT from pruning block data. | True |

> | consensus.evidence.MaxAgeDuration | 2 days | The maximum age of evidence before it is considered invalid. This value should be identical to the unbonding period. | True |

Reviewer: From this it remains unclear in which actual unit this param would be defined (IIRC it's seconds not days?).

Reviewer: ```suggestion
| consensus.evidence.MaxAgeDuration | 172800000000000 (2 days) | The maximum age (in nanoseconds) of evidence before it is considered invalid. This value should be identical to the unbonding period. | True |
```

Author: its defined a golang time.Duration which is always nanoseconds, so I just listed two days. If encoded properly, it can be represented as seconds I believe (like in other sdk modules), but since these are comet params they are not encoded normally.

will change to nano seconds

Author: this was also updated to what it will be for mainnet (21 days same as unbonding) in nanoseconds. The default will reflect that after #2417 and https://github.com/celestiaorg/cosmos-sdk/pull/345

> | consensus.evidence.MaxAgeDuration | 172800000000000 | The maximum age of evidence before it is considered invalid. This value should be identical to the unbonding period. | True |

Reviewer 2: What's the time unit here?

| consensus.evidence.MaxBytes | 1MB | Maximum size in bytes used by evidence in a given block. | True |
| consensus.validator.PubKeyTypes | Ed25519 | The type of public key used by validators. | False |
| consensus.Version.AppVersion | 1 | Determines protocol rules used for a given height. Incremented by the application upon an upgrade. | False |
| distribution.communitytax | 0.02 | Percentage of the inflation sent to the community pool. | True |
| distribution.WithdrawAddrEnabled | true | Enables delegators to withdraw funds to a different address. | True |
| distribution.BaseProposerReward | 0 | Reward for proposing a block. | True |
| distribution.BonusProposerReward | 0 | Extra reward for proposers based on the voting power included in the commit. | True |

> | gov.DepositParams.MinDeposit | 10000000utia | Minimum deposit for a proposal to enter voting period in seconds. | True |

Reviewer: This is very low compared to other chains IMO. We should increase this! IMO at least 100TIA better 1000TIA. Mainly to prevent spam proposals. cc @YazzyYaz @musalbas 
Can be discussed in a separate issue though.

Reviewer 2: Agree, also agree we can discuss it separately. First document current behavior.

Author: this is being updated in https://github.com/celestiaorg/celestia-app/pull/2417 and has been updated here, is okay to keep at 1000?

Reviewer: I think it's ok at 1000 for now.

| gov.DepositParams.MaxDepositPeriod |  | Maximum period for token holders to deposit on a proposal. | True |

> | gov.VotingParams.VotingPeriod |  | Duration of the voting period. | True |

Reviewer 2: Do we not know these? We should!

> | gov.VotingParams.VotingPeriod | 172800s | Duration of the voting period in seconds. | True |

Reviewer: ```suggestion
| gov.VotingParams.VotingPeriod | 172800 (2 days) | Duration of the voting period in seconds. | True |
```

| gov.TallyParams.Quorum | 33.4 | Minimum percentage of total stake needed to vote for a result to be considered valid. | True |
| gov.TallyParams.Threshold | 50.0 | Minimum proportion of Yes votes for proposal to pass. | True |
| gov.TallyParams.VetoThreshold | 33.4 | Minimum value of Veto votes to Total votes ratio for proposal to be vetoed. | True |
| slashing.SignedBlocksWindow | 5000 | The range of blocks used to count for downtime. | True |
| slashing.MinSignedPerWindow | 5 | Minumum signatures in the block. | True |
| slashing.DowntimeJailDuration | 10 mins | Duration of time a validator must stay jailed. | True |
| slashing.SlashFractionDoubleSign | 1/20 | Percentage slashed after a validator is jailed for downtime. | True |

> | slashing.SlashFractionDowntime | 1/100 | Percentage slashed after a validator is jailed for downtime. | True |

Reviewer 2: These aren't percentages.

| gov.DepositParams.MaxDepositPeriod | 172800 (2 days) | Maximum period for token holders to deposit on a proposal in seconds. | True |
| gov.VotingParams.VotingPeriod | 172800 (2 days) | Duration of the voting period in seconds. | True |
| gov.TallyParams.Quorum | 33.4% | Minimum percentage of total stake needed to vote for a result to be considered valid. | True |
| gov.TallyParams.Threshold | 50.0% | Minimum proportion of Yes votes for proposal to pass. | True |
| gov.TallyParams.VetoThreshold | 33.4% | Minimum value of Veto votes to Total votes ratio for proposal to be vetoed. | True |
| ibc.ClientGenesis.AllowedClients | []string{"06-solomachine", "07-tendermint"} | List of allowed IBC light clients. | True |
| ibc.ConnectionGenesis.MaxExpectedTimePerBlock | 7500000000000 (75 seconds) | Maximum expected time per block in nanoseconds under normal operation. | True |
| ibc.Transfer.SendEnabled | true | Enable sending tokens via IBC. | True |
| ibc.Transfer.ReceiveEnabled | true | Enable receiving tokens via IBC. | True |
| slashing.SignedBlocksWindow | 5000 | The range of blocks used to count for downtime. | True |

> | slashing.MinSignedPerWindow | 5 | Minumum signatures in the block. | True |

Reviewer: > Minumum signatures in the block.

The description here should be changed to sth more descriptive: Minumum number of blocks signed in the `SignedBlocksWindow`. Or sth like that?

Reviewer: cc @rootulp @evan-forbes

Reviewer 3: Created https://github.com/celestiaorg/celestia-app/issues/2469
